File: ios/learning/UdpCmd/UdpCmd/VolumeMessage.m
#import "VolumeMessage.h"

@implementation VolumeMessage

- (NSString *)toString {
    return [NSString localizedStringWithFormat:@"TYPE=%d,VOLUME=%d", self.cmdType, _volume];
}

- (VolumeMessage *)init:(int)volume {
    self = [super init];
    self.cmdType = VOLUME;
    [self setVolume:volume];
    return self;
}
@end

File: java/air-io/src/main/java/feuyeux/io/nio/udp/protocol/VolumeMessage.java
package feuyeux.io.nio.udp.protocol;

public class VolumeMessage extends BaseMessage {
    private int volume;

    public VolumeMessage() {
        cmdType = CMD_TYPE_QUERY_VOLUME;
    }

    public VolumeMessage(int volume) {
        cmdType = CMD_TYPE_QUERY_VOLUME;
        this.volume = volume;
    }

    public int volumeVal() {
        return volume;
    }

    @Override
    public String toString() {
        return "[QueryVolume] TYPE=" + cmdType + ",VOLUME=" + volume;
    }
}
